app/controllers/users_controller.rb

```diff
@@ -93,7 +93,11 @@ def update_language_settings
 
   def request_destroy
     set_tabs("account")
-    if @user.email?
+
+    if destroy_request_in_progress
+      flash[:settings_notice] = "You have already requested account deletion. Please, check your email for further instructions."
+      redirect_to "/settings/#{@tab}"
```

Comment:
Lately, we're making more an effort to switch to path helpers. Here it'd be 

```ruby
redirect_to user_settings_path(@tab)
```

if you want to update it (and the rest of the file/method). It's fine if you don't, I'll eventually get to it.